js/src/utils/toScopeState.test.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Internal dependencies
+ */
+import toScopeState from './toScopeState';
+
+const SCOPE = {
+	// Manage product listings and accounts for Google Shopping
+	CONTENT: 'https://www.googleapis.com/auth/content',
+	// Manage new site verifications with Google
+	SITE_VERIFICATION_VERIFY_ONLY:
+		'https://www.googleapis.com/auth/siteverification.verify_only',
+	// Manage AdWords campaigns
+	AD_WORDS: 'https://www.googleapis.com/auth/adwords',
+};
+
+describe( 'toScopeState', () => {
+	let contentScopes;
+	let siteVerificationScopes;
+	let gmcScopes;
+	let adsScopes;
+	let allScopes;
+
+	beforeEach( () => {
+		contentScopes = [ SCOPE.CONTENT ];
+		siteVerificationScopes = [ SCOPE.SITE_VERIFICATION_VERIFY_ONLY ];
+		gmcScopes = [ SCOPE.CONTENT, SCOPE.SITE_VERIFICATION_VERIFY_ONLY ];
+		adsScopes = [ SCOPE.AD_WORDS ];
+		allScopes = [
+			SCOPE.CONTENT,
+			SCOPE.SITE_VERIFICATION_VERIFY_ONLY,
+			SCOPE.AD_WORDS,
+		];
+	} );
+
+	it( 'should return an object structure containing three properties', () => {
+		const scopeState = toScopeState( false, [] );
+
+		expect( scopeState ).toHaveProperty( 'gmcRequired' );
+		expect( scopeState ).toHaveProperty( 'adsRequired' );
+		expect( scopeState ).toHaveProperty( 'glaRequired' );
+	} );
+
+	it( 'when the `scopes` parameter is not given, should get `false` for all results', () => {
+		expect( toScopeState( false ) ).toMatchObject( {
+			gmcRequired: false,
+			adsRequired: false,
+			glaRequired: false,
+		} );
+		expect( toScopeState( true ) ).toMatchObject( {
+			gmcRequired: false,
+			adsRequired: false,
+			glaRequired: false,
+		} );
+	} );
+
+	describe.each( [ [ false ], [ true ] ] )(
+		'For `gmcRequired`, if the parameter `adsSetupComplete` = %p',
+		( adsSetupComplete ) => {
+			it( 'and the `scopes` contains GMC required scopes, should be `true`', () => {
+				expect(
+					toScopeState( adsSetupComplete, gmcScopes ).gmcRequired
+				).toBe( true );
+			} );
+
+			it( "and the `scopes` doesn't contains GMC required scopes, should be `false`", () => {
```

Comment:
💅  ❔ I'd prefer to make the string connects with the `it` in front, so that it reads like "it returns false when...": 😄 

```suggestion
			it( "returns false when the `scopes` doesn't contains GMC required scopes", () => {
```

Reply:
Quoted from https://github.com/woocommerce/google-listings-and-ads/pull/834#issuecomment-874350184:

> 2\. 💅  Personally, I prefer the test descriptions where the conditions come first, and the expectation is last. I find it easier to write especially in nested `describe`s and more importantly easier to read when the full desc is build:
> 
> ```js
> describe( `When the path…`, () => {
> 	describe( 'for successful…', () => {
> 		it( 'should call …', () => {
> ```

This description structure is based on Tomek's suggestion as well, but I don't have a particular preference for it either. Maybe another 📜  for you and @tomalec. 😂

Reply:
That seems like two perspectives:
- Make the string connects with the `it` in front, so that it [code] reads like "it returns false when..."
- Make the result description build a story, so that test run/failure reads like "When the conditions are met, the subject should return false"

So would we target the code to be more readable, or the test result to be readable. Preferably, we would have both nice and readable :) I'm open to suggestions how to phrase that to make it so.

If we are not able to optimize for both, personally, I'd favor the latter, as the tests are meant to be written once and run every time. Then if they break, it would be easier to reason what and why was broken. Because when it breaks than most probably the person looking at the broken test result is the one who broke it, so the one who is not so familiar with how it should behave.